src/Shared/ActivatorUtilities/ActivatorUtilities.cs

```diff
@@ -47,13 +47,11 @@ public static object CreateInstance(IServiceProvider provider, Type instanceType
 
             ConstructorMatcher bestMatcher = default;
 
-            if (!instanceType.GetTypeInfo().IsAbstract)
+            if (!instanceType.IsAbstract)
             {
-                foreach (var constructor in instanceType
-                    .GetTypeInfo()
-                    .DeclaredConstructors)
+                foreach (var constructor in instanceType.GetConstructors())
                 {
-                    if (!constructor.IsStatic && constructor.IsPublic)
+                    if (!constructor.IsStatic)
```

Comment:
Note these changes match what was done in dotnet/runtimes ActivatorUtilities. See https://github.com/dotnet/runtime/pull/40227/files#diff-d04fd33a5eba20dc5b742e17b751d3b9c0860d114b73c777a0e140c1ceaccf6fR57

Reply:
Hmmmm when?

Reply:
Is `constructor.IsPublic` implied? Or does this change ActivatorUtilities.CreateInstance to use non-public ctors? If it's the latter, this seems like a fairly major breaking change.

Reply:
Previous conversation about this: https://github.com/dotnet/runtime/pull/40227/files#r463897549
TLDR: `GetConstructors` returns only public constructors

Reply:
Just looked again, it also only returns `Instance` constructors, so we could remove the `IsStatic` check from both copies (runtime and aspnetcore) of this code.

https://github.com/dotnet/runtime/blob/1821d9c14b970d58e0768256de138b6c0287e07d/src/libraries/System.Private.CoreLib/src/System/Type.cs#L175-L178